### ISSUE TITLE
when add segmentation result into feature map which is used to detect, my model didn't converge. 

### ISSUE THREAD

**Issue by moliqian** [BestSonny/SSTD#15]
@BestSonny Hi. 
Now, I train a net to segment and detect simultaneously on VOC2012. I use MaskResize and MaskPooling to fuse conv feature maps with segmentation result （1 - segment_result[0]）. But the model didin't converge.  
So, add BN layer into net  and  reduce learning rate to a small number. But the model didn't converge as well .
Have you met the problem when you trained your model? Could you give some advice ?
### Following is part of training log:
I0109 13:39:28.412003 21535 net.cpp:761] Ignoring source layer fc8
I0109 13:39:28.412008 21535 net.cpp:761] Ignoring source layer prob
I0109 13:39:29.777668 21535 solver.cpp:243] Iteration 0, loss = 26.1135
I0109 13:39:29.777715 21535 solver.cpp:259]     Train net output #0: mbox_loss = 22.8622 (* 1 = 22.8622 loss)
I0109 13:39:29.777726 21535 solver.cpp:259]     Train net output #1: seg_loss = 3.18038 (* 1 = 3.18038 loss)
I0109 13:39:29.777734 21535 sgd_solver.cpp:138] Iteration 0, lr = 0.001
I0109 13:40:46.876418 21535 solver.cpp:243] Iteration 10, loss = 16.8931
I0109 13:40:46.876471 21535 solver.cpp:259]     Train net output #0: mbox_loss = 14.9162 (* 1 = 14.9162 loss)
I0109 13:40:46.876482 21535 solver.cpp:259]     Train net output #1: seg_loss = 1.84232 (* 1 = 1.84232 loss)
I0109 13:40:46.876493 21535 sgd_solver.cpp:138] Iteration 10, lr = 0.001
I0109 13:41:01.436628 21535 solver.cpp:243] Iteration 20, loss = 16.8339
I0109 13:41:01.436683 21535 solver.cpp:259]     Train net output #0: mbox_loss = 15.0192 (* 1 = 15.0192 loss)
I0109 13:41:01.436695 21535 solver.cpp:259]     Train net output #1: seg_loss = 2.32088 (* 1 = 2.32088 loss)
I0109 13:41:01.436707 21535 sgd_solver.cpp:138] Iteration 20, lr = 0.001
I0109 13:41:15.868654 21535 solver.cpp:243] Iteration 30, loss = nan
I0109 13:41:15.868707 21535 solver.cpp:259]     Train net output #0: mbox_loss = nan (* 1 = nan loss)
I0109 13:41:15.868721 21535 solver.cpp:259]     Train net output #1: seg_loss = 87.3365 (* 1 = 87.3365 loss)

**Comment by BestSonny** [comment]
@moliqian  Your data or network may have some unknown bugs
